digitalpalireader/content/js/dict_xml.js

```diff
@@ -1,796 +1,634 @@
+
+	if(G_dppn.length == 0) {
+		for (i in D) {
+			for (j in D[i]) {
+				G_dppn.push([i,D[i][j]]);
+			}
+		}
+	}
+	for (i in G_dppn) {
+		if(tname) {
+			nname = "'"+toUni(G_dppn[i][0])+'/'+G_dppn[i][1]+"','"+toUni(G_dppn[i][0])+"'";
+			break;
+		}
+		if (G_dppn[i][0] == tloc[0] && G_dppn[i][1] == tloc[1]+'/'+tloc[2]) {
+			tname = "'"+toUni(G_dppn[i][0])+'/'+G_dppn[i][1]+"','"+toUni(G_dppn[i][0])+"'";
+		}
+		else lname = "'"+toUni(G_dppn[i][0])+'/'+G_dppn[i][1]+"','"+toUni(G_dppn[i][0])+"'";
+	}
+	if (!tname) lname = null;
+	// buttons
+
+	var tout = '';
+
+	if (G_dppnhist.length > 1) { // show select
+		var showing = '<select title="go to history" onchange="if(this.selectedIndex != 0) { G_dhmark=this.length-1-this.selectedIndex; DPPNXML(this.options[this.selectedIndex].value,1);}"><option>- history -</option>';
+		for (i = G_dppnhist.length-1; i >= 0; i--) {
+			showing += '<option value="'+G_dppnhist[i]+'"';
+			if (i == G_dhmark) { showing += ' selected'; }
+			var dhs = G_dppnhist[i].split(',');
+			showing += '>' + (dhs[1] ? dhs[1] : dhs[0]) + '</option>';
+		}
+		showing += '</select>';
+		tout += (tout.length > 0 ? ' ' : '') + showing;
+
+	}
+
+	if (lname) tout += '<span class="abut lbut tiny" onclick="DPPNXML('+lname+')" />&lt;</span>';
+	if (nname) tout += '<span class="abut rbut tiny" onclick="DPPNXML('+nname+')" />&gt;</span>';
+
+
+	$('#difhist').html('<table><tr><td>' + tout + '</td></tr></table>');
+	if(document.getElementById('bottom')) {
+		document.getElementById('cdif').scrollTop=0;
+		DPRBottomPaneUpdateStyle();
+	}
+	else document.getElementById('dictc').scrollTop=0;
+
+	// permalink
+
+	if(DPR_PAL.isDictionaryFeature()) {
+		var permalink = document.location.href;
+		if(/\&entry=/.exec(permalink)) {
+			permalink = permalink.replace(/&entry=[^&]*/,'&entry='+toVel(filein));
+		}
+		else permalink += '&entry='+toVel(filein);
+		try {
+			window.history.replaceState('Object', 'Title', permalink);
+		}
+		catch(ex) {
+		}
+	}
+
 }
 
-var G_skt = []
-var G_skthist = []
-var G_shmark = 0
+var G_skt = [];
+var G_skthist = [];
+var G_shmark = 0;
+
+function sktRXML(no,add)
+{
+	if(add == 'right') return;
+	if(add == true) {
+		//sendDPPNXML(toVel(filein),true);
+		//return;
+	}
+
+	//moveFrame(1);
+
+	clearDivs('dif');
+
+	if(/[^0-9]/.test(no)) {
+		for(var i = 0; i < sktR.length;i++) {
+			if (sktR[i] == no) {
+				no = i;
+				break;
+			}
+		}
+		if(/[^0-9]/.test(no))
+			return;
+	}
+
+	// xml
+
+	var file = DPR_PAL.baseUrl + 'sanskrit/content/skt-roots/'+sktR[no]+'.htm';
+	var xmlhttp = new window.XMLHttpRequest();
+	xmlhttp.open("GET", file, false);
+	xmlhttp.send(null);
+	var xmlDoc = xmlhttp.responseXML.documentElement;
+	var s = new XMLSerializer();
+	var data = s.serializeToString(xmlDoc);
+
+	data = data.replace(/<p><a href="index.htm">Index<\/a><\/p>/,'').replace(/<a href="([^.]+)[^>]+/g,"<a class=\"green\" href=\"javascript:void(0)\" onclick=\"sktRXML('$1')\"");
+
+	// output
+
+    displayDictData(data);
```

Comment:
Clicking on a sanskrit root result item does not work locally in Chrome and Firefox. For loading `html` files, for me only an asynchronous request such below works, but there might be easier workarounds than this.

```  
  var data;
  var file = DPR_PAL.baseUrl + 'sanskrit/content/skt-roots/'+sktR[no]+'.htm';
  var xmlhttp = new window.XMLHttpRequest();
  var xmlhttp = new XMLHttpRequest();
  xmlhttp.onload = function () {
    var s = new XMLSerializer();
    var xmlDoc = xmlhttp.responseXML.documentElement;
    data = s.serializeToString(xmlDoc);
    data = data.replace(/<p><a href="index.htm">Index<\/a><\/p>/,'').replace(/<a href="([^.]+)[^>]+/g,"<a class=\"green\" href=\"javascript:void(0)\" onclick=\"sktRXML('$1')\"");
    displayDictData(data);
  }
  xmlhttp.onerror = function () {
    dump('Error while getting XML.');
  }
  xmlhttp.open('GET', file);
  xmlhttp.responseType = 'document';
  xmlhttp.send()0;
```

Reply:
yeah html cannot be loaded by sync requests - this is as per the API documentation. 
i renamed the file to xml and it was working fine.

that is the fix i am working on currently - ie renaming all files to xml